armadillo/src/main/java/org/molgenis/armadillo/settings/UserDetails.java

```diff
@@ -0,0 +1,55 @@
+package org.molgenis.armadillo.settings;
+
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+// for generation to JSON
+public class UserDetails {
```

Comment:
We use `@AutoValue` for objects that need to be serialised to JSON. See for examples `Workspace` or `RPackage`